### PR TITLE
Make FSharp.Build build netstandard2.0

### PR DESCRIPTION
**Pull request by KevinRansom**
There is no good reason for not building this netstandard2.0. 

So making it happen.

### CODE REVIEW

File: proto.proj [hunk 1]
```diff
@@ -6,9 +6,6 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <Projects Include="src\fsharp\FSharp.Build\FSharp.Build.fsproj">
-      <AdditionalProperties Condition="'$(OS)' == 'Unix'">TargetFramework=netcoreapp3.1</AdditionalProperties>
-    </Projects>
     <Projects Include="src\fsharp\fsc\fsc.fsproj">
       <AdditionalProperties Condition="'$(OS)' == 'Unix'">TargetFramework=netcoreapp3.1</AdditionalProperties>
```
Comment (cartermp) [inline proto.proj:10]: can this now be moved to the upper property group unconditionally?

Comment (cartermp) [reply proto.proj:10]: hmm, this is proto.proj, was this intended?

Comment (KevinRansom) [reply proto.proj:13]: it was

File: src/fsharp/FSharp.Build/FSharp.Build.fsproj [hunk 2]
```diff
@@ -4,15 +4,15 @@
 
   <PropertyGroup>
     <OutputType>Library</OutputType>
-    <TargetFrameworks Condition="'$(ProtoTargetFramework)' != ''">$(ProtoTargetFramework)</TargetFrameworks>
-    <TargetFrameworks Condition="'$(ProtoTargetFramework)' == ''">net472;netcoreapp3.1</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' == 'Unix'">netcoreapp3.1</TargetFrameworks>
+    <TargetFramework Condition="'$(Configuration)' != 'Proto'">netstandard2.0</TargetFramework>
+    <TargetFrameworks Condition="'$(Configuration)' == 'Proto'">netstandard2.0</TargetFrameworks>
```
Comment (brettfo) [inline src/fsharp/FSharp.Build/FSharp.Build.fsproj:8]: These should be collapsed into a single `<TargetFramework>netstandard2.0</TargetFramework>`.

File: src/fsharp/FSharp.Build/FSharp.Build.fsproj [hunk 3]
```diff
@@ -49,10 +49,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Build" Version="$(MicrosoftBuildVersion)" />
-    <PackageReference Include="Microsoft.Build.Framework" Version="$(MicrosoftBuildFrameworkVersion)" />
-    <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(MicrosoftBuildTasksCoreVersion)" />
-    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="$(MicrosoftBuildUtilitiesCoreVersion)" />
+    <PackageReference Include="Microsoft.Build" Version="$(MicrosoftBuildFrameworkVersion)" PrivateAssets="all" />
```
Comment (brettfo) [inline src/fsharp/FSharp.Build/FSharp.Build.fsproj:52]: Keep the package name and variable name in sync.

File: src/fsharp/Microsoft.FSharp.Compiler/Microsoft.FSharp.Compiler.csproj [hunk 4]
```diff
@@ -17,7 +17,7 @@
       <AdditionalProperties>TargetFramework=netcoreapp3.1</AdditionalProperties>
     </ProjectReference>
     <ProjectReference Include="..\FSharp.Build\FSharp.Build.fsproj">
-      <AdditionalProperties>TargetFramework=netcoreapp3.1</AdditionalProperties>
+      <AdditionalProperties>TargetFramework=netstandard2.0</AdditionalProperties>
```
Comment (brettfo) [inline src/fsharp/Microsoft.FSharp.Compiler/Microsoft.FSharp.Compiler.csproj:20]: If the TFM of FSharp.Build is only netstandard2.0, this shouldn't be necessary.

File: src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj [hunk 5]
```diff
@@ -5,7 +5,6 @@
   <PropertyGroup>
     <OutputType>Library</OutputType>
     <TargetFrameworks>net472;netstandard2.0</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' == 'Unix'">netstandard2.0</TargetFrameworks>
```
Comment (brettfo) [inline src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj:8]: Doesn't this cause it to fail on mac/linux?

**Review by brettfo — CHANGES_REQUESTED** [review]

File: src/fsharp/FSharp.Build/FSharp.Build.fsproj (hunk 2 above)
Comment (KevinRansom) [reply src/fsharp/FSharp.Build/FSharp.Build.fsproj:8]: @brettfo , yeah ... no.  We publish and publish requires either a single tfm or a specified framework, I couldn't figure out how to specify a framework on the unix build.
A single tfm in the product build fails with missing tfms.   I wasted days trying to figure out something that would actually work.  But we can get together and you can experience the frustration of it if you would like,

File: src/fsharp/FSharp.Build/FSharp.Build.fsproj (hunk 3 above)
Comment (KevinRansom) [reply src/fsharp/FSharp.Build/FSharp.Build.fsproj:52]: ooo good eyes.

File: src/fsharp/Microsoft.FSharp.Compiler/Microsoft.FSharp.Compiler.csproj (hunk 4 above)
Comment (KevinRansom) [reply src/fsharp/Microsoft.FSharp.Compiler/Microsoft.FSharp.Compiler.csproj:20]: Yeah you would think .... however ... the proto project is a bit wierder than it looks, like above we can get together and you can experience the wierdness if you like.

File: src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj (hunk 5 above)
Comment (KevinRansom) [reply src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj:8]: it was a mistake, I need to revert it.

Comment (KevinRansom) [reply src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj:8]: Although no, it doesn't and I don't really understand why.

Comment (KevinRansom) [reply src/fsharp/FSharp.Compiler.Interactive.Settings/FSharp.Compiler.Interactive.Settings.fsproj]: @brettfo, you can build net472 on linux.  You can't run it, but you can for sure build it.

**Review by brettfo — APPROVED** [review]

**Comment by KevinRansom** [comment]
Thanks brett, I appreciate it.  I wish our build were simpler.

**Pull request merged**